server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorDVLeafFieldData.java

```diff
@@ -76,4 +87,109 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
         }
     }
 
+    @Override
+    public FormattedDocValues getFormattedValues(DocValueFormat format) {
+
+        return switch (elementType) {
+            case BYTE, BIT -> new FormattedDocValues() {
+                final DenseVectorNumericByteValues values = getByteValues();
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    return values.advanceExact(docId);
+                }
+
+                @Override
+                public int docValueCount() throws IOException {
+                    return values.docValueCount();
+                }
+
+                @Override
+                public Object nextValue() throws IOException {
+                    return values.nextValue();
+                }
+            };
+            case FLOAT -> new FormattedDocValues() {
+                final DenseVectorNumericFloatValues values = getFloatValues();
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    return values.advanceExact(docId);
+                }
+
+                @Override
+                public int docValueCount() throws IOException {
+                    return values.docValueCount();
+                }
+
+                @Override
+                public Object nextValue() throws IOException {
+                    return values.nextValue();
+                }
+            };
+        };
+    }
+
+    private DenseVectorNumericByteValues getByteValues() {
+        int dims = elementType == ElementType.BIT ? this.dims / Byte.SIZE : this.dims;
+        return new DenseVectorNumericByteValues(dims) {
+            @Override
+            public boolean advanceExact(int docID) throws IOException {
+                if (indexed) {
+                    final ByteVectorValues byteVectorValues = reader.getByteVectorValues(field);
+                    if (byteVectorValues.advance(docID) == NO_MORE_DOCS) {
```

Comment:
This doesn't compile. Now we need to access vectors through iterator, as in [this example](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/script/field/vectors/ByteKnnDenseVectorDocValuesField.java#L50)